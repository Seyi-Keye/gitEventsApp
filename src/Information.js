import React, { Component } from "react";
import moment from "moment";

class Information extends Component {
  render() {
    const { data } = this.props;
    const type = data.type || "n/a";
    const date = moment(data.created_at).toString() || "n/a";
    const name = (data.actor && data.actor.display_login) || "n/a";
    return (
      <div
        style={{
          color: "white",
          marginTop: "20px",
          padding: "10px"
        }}
      >
        {data ? `${type} created on ${date} by ${name}` : ""}
      </div>
    );
  }
}
export default Information;
