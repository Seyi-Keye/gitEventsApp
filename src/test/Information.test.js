import React from "react";
import Information from "../Information";
import renderer from "react-test-renderer";

describe("Information component", () => {
  let wrapper = renderer.create(<Information data={{}} />);
  it("renders succesfully", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders without props", () => {
    const testInstace = wrapper.root;
    expect(testInstace.props).toEqual({ data: {} });
  });
});

describe("Information component", () => {
  const testProps = {
    data: {
      type: "Test Type",
      created_at: "2018-12-17T03:24:00",
      actor: {
        display_login: "Aromokeye"
      }
    }
  };
  let wrapper = renderer.create(<Information data={testProps.data} />);

  it("renders succesfully with props", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders with props", () => {
    const testInstace = wrapper.root;
    console.log(testInstace.props.data);
    expect(testInstace.props.data).toEqual(testProps.data);
  });
});
