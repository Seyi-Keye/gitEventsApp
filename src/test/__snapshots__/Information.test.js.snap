// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information component renders succesfully 1`] = `
<div
  style={
    Object {
      "color": "white",
      "marginTop": "20px",
      "padding": "10px",
    }
  }>
  n/a created on Sun May 13 2018 15:45:02 GMT+0100 by n/a
</div>
`;

exports[`Information component renders succesfully with props 1`] = `
<div
  style={
    Object {
      "color": "white",
      "marginTop": "20px",
      "padding": "10px",
    }
  }>
  Test Type created on Mon Dec 17 2018 03:24:00 GMT+0100 by Aromokeye
</div>
`;
